frontend/src/App.tsx

```diff
@@ -1,41 +1,21 @@
 import "./App.css";
 import React, { useEffect, useState } from "react";
 import { ProductList } from "./components/ProductsList.component";
-import data from "./db/db.json";
 import { Product } from "./model/model";
 
 const App: React.FC = () => {
-    useEffect(() => {
-        fetch("https://jsonplaceholder.typicode.com/posts", {
-            method: "POST",
-            body: JSON.stringify(data),
-            headers: {
-                "Content-type": "application/json; charset=UTF-8",
-            },
-        })
-            .then((response) => {
-                if (!response.ok) {
-                    throw new Error("Failed to post data");
-                }
-                return response.json();
-            })
-            .catch((error) => console.error("Error posting data:", error));
-    }, []);
-
     const [products, setProducts] = useState<Product[]>([]);
 
     useEffect(() => {
-        fetch("http://localhost:3000/products")
-            .then((response) => {
-                if (!response.ok) {
-                    throw new Error("Failed to fetch data");
-                }
-                return response.json();
-            })
-            .then((json) => {
-                setProducts(json);
-            })
-            .catch((error) => console.error("Error fetching data:", error));
+        async function fetchData() {
+            const response = await fetch("http://localhost:3000/api/products");
+            if (!response.ok) {
```

Comment:
The reason I used the response.ok property instead of catch block is that fetch() doesn't throw an error when the server returns a bad HTTP status. It rejects only if a request cannot be made or a response cannot be retrieved. The response.ok property allows to distinguish good from bad HTTP response statuses. The property is set to true only if the response has status 200-299.